sensor/common/deduper/deduper.go

```diff
@@ -58,24 +61,35 @@ var (
 
 // deduper takes care of deduping sensor events.
 type deduper struct {
-	stream   messagestream.SensorMessageStream
-	lastSent map[Key]uint64
+	stream         messagestream.SensorMessageStream
+	lastSent       map[deduperkey.Key]uint64
+	observationSet *ClosableSet
```

Comment:
This is a mega nit so feel free to ignore, but I think it should be Closeable

Reply:
Done